ami_assets/usr/lib/systemd/hugetlb-reserve-pages.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+#
+# Reserve hugepages when NUMA-aware.
+# Always reserve pages only on the first NUMA node.
+#
+# N.B. This currently only supports one or two NUMA nodes as it does not
+# NUMA nodes aren't guaranteed to be numbered contiguously and fully parsing
+# /sys/devices/system/node/online is non-trivial.
+#
+# NUMPAGES should be replaced with the required number of pages before use.
+nodes_path=/sys/devices/system/node/
+if [ ! -d "$nodes_path" ]; then
+	echo "ERROR: $nodes_path does not exist"
+	exit 1
+fi
+
+reserve_pages()
+{
+	echo "$1" > "$nodes_path/node$2/hugepages/hugepages-1048576kB/nr_hugepages"
+}
+
+# Get the number of NUMA nodes.
+numa_node_count=$(lscpu | grep -F "NUMA node(s)" | awk '{ print $3 }')
+
+reserve_pages NUMPAGES 0
```

Comment:
Isn't `NUMPAGES` an env var?

```suggestion
reserve_pages "$NUMPAGES" 0
```

Reply:
Ah no, I see now :)

Maybe it should be an env var, or an argument to the script? (given that this is a OneShot script rather than a tuned configuration file)